### PR TITLE
Validate student email address is different than parent/guardian's email address

### PR DESCRIPTION
**Pull request by shaunxp20**
I added a custom validator to ensure that the student's email address is different than the parent/guardian's email address.

Repro steps are in the issue #2347

### CODE REVIEW

**Comment by shaunxp20** [comment]
@stenington I am still going to do more QA on this locally (later today), I just wanted get a PR up for you to look at sooner rather than later. 😅 It makes me nervous touching `profile_controller.rb`. 😬 😬 😅

File: app/controllers/concerns/profile_controller.rb [hunk 1]
```diff
@@ -41,6 +41,8 @@ def update
       else
         render "passwords/edit"
       end
+    elsif profile.errors["account.email"].any?
+      render 'email_addresses/edit'
     elsif profile.errors["account.city"].any? or
             profile.errors["account.state_province"].any? or
               profile.errors["account.country"].any?
```
Comment (shaunxp20) [inline app/controllers/concerns/profile_controller.rb]: Btw, my goal was to test each branch of the conditionals here for students, but I couldn't find a way to get any (validation) errors for `city`, `state_province`, or `country`, I couldn't even find any validators for these fields, anywho tldr; I'm not sure how to reach this branch! 🤷‍♂ 🌳 🚶 🎶

Comment (stenington) [reply app/controllers/concerns/profile_controller.rb]: Yeah, they may be unreachable. I've encountered a fair amount of leftover stuff around locations.

**Comment by shaunxp20** [comment]
> I'm good with the PR, if work is done. 👍

Cool, at this point, I think a good next step would be to put it on staging for more QAing. 😬 😅

**Pull request merged**